src/com/sun/jna/Function.java

```diff
@@ -274,12 +285,15 @@ public Object invoke(Class returnType, Object[] inArgs) {
     public Object invoke(Class returnType, Object[] inArgs, Map options) {
         // Clone the argument array to obtain a scratch space for modified
         // types/values
-        Object[] args = { };
+        if (argTable.get() == null) {
+          createArgTable();
+        }
```

Comment:
Move this code to the _ThreadLocal_ (and make the instance _final_) as follows:
```java
private static final ThreadLocal<Object[][]> argTable = new ThreadLocal<Object[][]>() {
    @Override
    protected Object[][] initialValue() {
        Object args[][] = new Object[MAX_NARGS+1][];  //+1 for zero args
        for(int a=0;a<=MAX_NARGS;a++) {
            args[a] = new Object[a];
        }
        return args;
    }
};
```
and dispense with _createArgsTable_. The way _ThreadLocal.get()_ works is *exactly* equivalent to this code, only cleaner